contracts/Colony.sol

```diff
@@ -125,11 +127,36 @@ contract Colony is ColonyStorage {
   function getDomainCount() public view returns (uint256) {
     return domainCount;
   }
-
   function setFunctionReviewers(bytes4 _sig, uint8 _firstReviewer, uint8 _secondReviewer)
   private
   {
     uint8[2] memory _reviewers = [_firstReviewer, _secondReviewer];
     reviewers[_sig] = _reviewers;
   }
+
+  modifier verifyKey(bytes key) {
+    uint256 colonyAddress;
+    uint256 skillid;
+    uint256 userAddress;
+    assembly {
+        colonyAddress := mload(add(key,32))
+        skillid := mload(add(key,52)) // Colony address was 20 bytes long, so add 20 bytes
+        userAddress := mload(add(key,84)) // Skillid was 32 bytes long, so add 32 bytes
+    }
+    colonyAddress >>= 96;
+    userAddress >>= 96;
```

Comment:
fyi there appears to be an assembly level shift right operator `sar(x, y)`

Reply:
Looking at the documentation, we would want `shr` not `sar`, but that's good to know. There might be a little gas-gain here, but I'd prefer to use as little assembly as possible in general.